<launch>

    <arg name="ctrls_namespace" default="z_baxter_jar" />

    <arg name="marker_size" default="4.0" />
    <arg name="max_new_marker_error" default="0.08" />
    <arg name="max_track_error" default="0.2" />


    <group>
        <group if="$(eval arg('run_AR_tracker') == true)">
            <node name="lh_ar_track" pkg="ar_track_alvar" type="individualMarkersNoKinect" respawn="false" output="screen">
                <param name="marker_size"           value="$(arg marker_size)" />
                <param name="max_new_marker_error"  value="$(arg max_new_marker_error)" />
                <param name="max_track_error"       value="$(arg max_track_error)" />
                <param name="output_frame"          value="/base" />

                <remap from="camera_image"          to="/cameras/left_hand_camera/image" />
                <remap from="camera_info"           to="/cameras/left_hand_camera/camera_info" />
            </node>
        </group>

        <node name="motion_controller" pkg="me495_baxter_jar" type="motionControl.py" respawn="true" output="screen"/>
        <node name="gripper_controller" pkg="me495_baxter_jar" type="gripperControl.py" respawn="true" output="screen"/>
        <node name="pose_relay" pkg="me495_baxter_jar" type="relayObjPose.py" respawn="true" output="screen"/>
    </group>

    <group if="$(eval arg('run_sequence') == true)">
            <node name="director" pkg="me495_baxter_jar" type="sequencer.py" respawn="false" output="screen" />
    </group>

</launch>